app/controllers/dashboard_controller.rb

```diff
@@ -5,14 +5,20 @@ class DashboardController < ApplicationController
   skip_verify_authorized only: :show
 
   def show
-    # This is going to be slow because it makes a call to DSA for each work.
-    # It is also going to block rendering since it is synchronous.
+    # This is going to be slow because it is going to block rendering since it is synchronous.
     # Also, it processes all of a user's works, but not all may be rendered on the dashboard.
+    @druid_to_status_map = Sdr::Repository.statuses(druids: druids_for_works_for_current_user)
     @draft_works = []
     @status_map = current_user.works.to_h do |work|
-      version_status = work.druid ? Sdr::Repository.status(druid: work.druid) : VersionStatus::NilStatus.new
+      version_status = @druid_to_status_map[work.druid] || VersionStatus::NilStatus.new
       @draft_works << work if version_status.draft?
       [work.id, version_status]
     end
   end
+
+  private
+
+  def druids_for_works_for_current_user
+    current_user.works.where.not(druid: nil).pluck(:druid)
```

Comment:
Maybe we'll eventually replace this with a policy scope? (But this is fine for now, I think.)